pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -956,6 +955,10 @@ func (cb *ClusterBuilder) applyConnectionPool(mesh *meshconfig.MeshConfig, mc *M
 		}
 	}
 
+	if cb.applyH2Upgrade(mc, port, mesh, settings) {
+		return
```

Comment:
when it is upgraded, no `UseClientProtocol`. In validation, we also do not allow setting both